### ISSUE TITLE
Interface binding

### ISSUE THREAD

**Issue by evacomm** [infusion/node-dhcp#37]
For a DHCP server, it would be ideal to be able to bind an instance to an interface . (or perhaps a group of interfaces).  If you were building a router, you only wan DHCP offered on LAN ports, not the WAN ports.

Thanks for taking the time to build this project, it's been very helpful!

**Comment by nistorj** [comment]
+1

**Comment by infusion** [comment]
Did you guys try to pass a host you want to listen on to the `listen`-method like so?

```
// dhcp.listen(port, host, callback)
dhcp.listen(null, '192.168.3.0');
```

**Comment by nistorj** [comment]
Ah yes, thanks!

I edited lib/node_modules/dhcp/bin/dhcpd-cli.js and ended up with:
//server.listen();
server.listen(null,'10.10.10.1');


root@guestshell ~/Python_DHCP: netstat -an | grep :67
udp        0      0 0.0.0.0:67              0.0.0.0:*
root@guestshell ~/Python_DHCP: netstat -an | grep :67
udp        0      0 10.10.10.1:67        0.0.0.0:*
root@guestshell ~/Python_DHCP:

thanks!

**Comment by infusion** [comment]
Yeah! You're welcome :) Do you have any suggestions to make things better for others too?

**Comment by infusion** [comment]
I quickly added the bind option to the CLI daemon. Could you check it out, if that's the way you would like to have it?

**Comment by nistorj** [comment]
slick.  Works well for IP addr yes, thanks!

For binding to an interface name (such as Vlan400, or Eth1/x) it doesn't but that's ok for me, I can work with it.   Thanks so much!

**Comment by infusion** [comment]
It's typically IP based, since its the only way to make it work across platforms, especially Windows. Under Linux you need to use SO_BINDTODEVICE, under OSX you need to use IP_BOUND_IF, but as far as I know there is no option to pass this info to the OS using nodejs.
